test/e2e/common/server.test.ts

```diff
@@ -134,9 +138,13 @@ describe("common/server", () => {
     });
 
     await setupPubSubSubscriptions();
+    // Sleep for 2 seconds to allow all subscriptions to be registered
+    await NodeJSUtil.sleep(2000);
```

Comment:
Is this really required? When we `register` a subscription, it just gets added to an internal data structure. Only after the server starts below, and Dapr sidecar calls `/dapr/subscribe` on app port, we return the list of subscriptions to the runtime which registers the subscriptions actually.

Reply:
That is interesting why this works, do you mind running this locally and see logs (with timestamps) on both gRPC and HTTP based sidecars? I think that should give us more idea.

Reply:
Resolved this with the comment above.